### DISCUSSION TITLE
DHT20 ESP32 NODEMCU

### DISCUSSION THREAD

**Discussion by miwi17null6** [micropython/micropython#10693] [ESP32] [answered]
Hello,

I flashed an esp32 nodemcu with micropython. a dht20 is to be used here. I connected it correctly, but I can't get the program to run. the i2c bus is recognized because it outputs the address 0x38.

i installed the dht20 library via tonny ide. (https://pypi.org/project/dht20-sensor/)

`from dht20_sensor.sensor import DHT20Sensor`

But there is always the error that the module could not be found.

how can i get the library on the esp32?

or to put it another way, does the dht20 currently work with the esp32?

**Comment by davefes** [comment]
Maybe check these out:
https://github.com/flrrth/pico-dht20
http://47.106.166.129/Embeded/pico-micropython-grove/blob/master/I2C/dht20.py

**Reply by miwi17null6** [reply]
https://github.com/flrrth/pico-dht20 
I had already tried. but there is always a syntax error in line 99 dht20.py

http://47.106.166.129/Embeded/pico-micropython-grove/blob/master/I2C/dht20.py
seems to work. but how do I get the values from the class DHT20?
I'm still inexperienced in this regard, but I want to learn more...

**Reply by davefes** [reply]
> I had already tried. but there is always a syntax error in line 99 dht20.py

Probably means there is an error in the line before, but I can't guess what it is.  Copy the exact error to your next response.

> but how do I get the values from the class DHT20?
```py
temperature = dht20_temperature()
print (f'{temperature}')
```
should do it.

**Reply by miwi17null6** [reply]
I'm using this script right now http://47.106.166.129/Embeded/pico-micropython-grove/blob/master/I2C/dht20.py
I have a boot.py and a dht20.py on the ESP32
boot.py:
```python
from machine import Pin, I2C
from utime import sleep

from dht20 import DHT20

temperature = dht20_temperature()
print (f'{temperature}')       #line 7
```
output:
```python
Traceback (most recent call last):
  File "boot.py", line 7
SyntaxError: invalid syntax
MicroPython v1.13 on 2020-09-02; ESP32 module with ESP32
Type "help()" for more information.
```

if I delete line 7, the next error comes

```python
Traceback (most recent call last):
  File "boot.py", line 6, in <module>
NameError: name 'dht20_temperature' isn't defined
MicroPython v1.13 on 2020-09-02; ESP32 module with ESP32
Type "help()" for more information.
```
I've been trying for a while now to get the values, but there are only errors. Can't really find any instructions for it. Anyone an idea?

**Reply by davefes** [reply]
At the REPL prompt `>>>` type `help('modules')` and see if DHT20 is in the list.

My v1.19.1 shows `dht` so try just `import dht` and see what happens.
After it is imported doing `dir('dht')` at the prompt does not show a DHT20.

https://docs.micropython.org/en/latest/esp8266/tutorial/dht.html
mentioned the DHT11 and DHT22.
```py
import dht
import machine

d = dht.DHT11(machine.Pin(4))

import dht

import machine

d = dht.DHT22(machine.Pin(4))
```
But it looks like it is not in the ESP32 port.  And there is nothing at https://awesome-micropython.com/

**Comment by miwi17null6** [comment, marked as answer]
It works!

I found this: https://github.com/mcauser/micropython-dht12 and using the script from here: http://47.106.166.129/Embeded/pico-micropython-grove/blob/master/I2C/dht20.py rewritten.

**Reply by miwi17null6** [reply]
```python
import time
import dht20
from machine import I2C, Pin
i2c = I2C(scl=Pin(22), sda=Pin(21))
sensor = dht20.DHT20(i2c)

while True:
    sensor.read_dht20()
    print(sensor.dht20_temperature())
    print(sensor.dht20_humidity())
    time.sleep_ms(4000)
```

**Comment by CStrangi08** [comment]
What library did you install to import dht20?
